File: perumarket-erp-backend/perumarket-erp-backend/src/main/java/com/perumarket/erp/models/entity/Ruta.java
package com.perumarket.erp.models.entity;

import jakarta.persistence.*;
import lombok.Data;
import java.math.BigDecimal;
import java.time.LocalDateTime;

import org.hibernate.annotations.CreationTimestamp;
import org.hibernate.annotations.UpdateTimestamp;

@Data
@Entity
@Table(name = "ruta")
public class Ruta {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    private String nombre;
    private String origen;
    private String destino;

    @Column(name = "distancia_km")
    private BigDecimal distanciaKm;

    @Column(name = "tiempo_estimado_horas")
    private BigDecimal tiempoEstimadoHoras;

    @Column(name = "costo_base")
    private BigDecimal costoBase;

    @CreationTimestamp
    private LocalDateTime fechaCreacion;

    @UpdateTimestamp
    private LocalDateTime fechaActualizacion;
}

File: perumarket-erp-frontend/src/types/rutas/ruta.ts
// src/types/ruta.ts
export interface Ruta {
  id?: number; 
  nombre: string;
  origen: string;
  destino: string;
  distancia_km?: number | null;
  tiempo_estimado_horas?: number | null;
  costo_base?: number | null;
}
export interface CrearRutaDTO {
  nombre?: string;
  origen?: string;
  destino?: string;
  distancia_km?: number | null;
  tiempo_estimado_horas?: number | null;
  costo_base?: number | null;
}
